runner_service/utils.py

```diff
@@ -206,18 +207,21 @@ def timeout_handler():
             raise SSHTimeout
 
         socket.setdefaulttimeout(self.timeout)
-        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         try:
-            s.connect((self.host, self.port))
+            family, type, proto, canonname, sockaddr = socket.getaddrinfo(self.host, self.port, 0, 0, socket.SOL_TCP)[0]
         except socket.gaierror:
             raise HostNotFound
-        except ConnectionRefusedError:
-            raise SSHNotAccessible
-        except socket.timeout:
-            raise SSHTimeout
-        else:
-            s.shutdown(socket.SHUT_RDWR)
-            s.close()
+
+        with socket.socket(family, socket.SOCK_STREAM) as s:
+            try:
+                s.connect(sockaddr)
+            except ConnectionRefusedError:
+                raise SSHNotAccessible
+            except socket.timeout:
+                raise SSHTimeout
+            else:
+                s.shutdown(socket.SHUT_RDWR)
+                s.close()
```

Comment:
Since we're using the context manager we don't need to close the socket ourselves anymore.